compiler/noirc_evaluator/src/ssa/opt/die.rs

```diff
@@ -599,16 +623,26 @@ struct RcTracker {
     // We also separately track all IncrementRc instructions and all array types which have been mutably borrowed.
     // If an array is the same type as one of those non-mutated array types, we can safely remove all IncrementRc instructions on that array.
     inc_rcs: HashMap<ValueId, HashSet<InstructionId>>,
-    // When tracking mutations we consider arrays with the same type as all being possibly mutated.
-    mutated_array_types: HashSet<Type>,
+    // Mutated arrays shared across the blocks of the function.
+    mutated_array_types: &'a mut HashSet<Type>,
     // The SSA often creates patterns where after simplifications we end up with repeat
     // IncrementRc instructions on the same value. We track whether the previous instruction was an IncrementRc,
     // and if the current instruction is also an IncrementRc on the same value we remove the current instruction.
     // `None` if the previous instruction was anything other than an IncrementRc
     previous_inc_rc: Option<ValueId>,
 }
 
-impl RcTracker {
+impl<'a> RcTracker<'a> {
+    fn new(mutated_array_types: &'a mut HashSet<Type>) -> Self {
+        Self {
+            rcs_with_possible_pairs: Default::default(),
+            rc_pairs_to_remove: Default::default(),
+            inc_rcs: Default::default(),
+            previous_inc_rc: Default::default(),
+            mutated_array_types,
+        }
+    }
+
     fn mark_terminator_arrays_as_used(&mut self, function: &Function, block: &BasicBlock) {
```

Comment:
https://github.com/AztecProtocol/aztec-packages/pull/11294/commits/ac9e584df4b7e533f02d11178968774f3a523e35

Reply:
This is superceded by https://github.com/AztecProtocol/aztec-packages/commit/4608c490032f2467cead6b76f4e81f9a4bf9ab50